lua/soulbound_abilities/beebarrel.lua

```diff
@@ -0,0 +1,136 @@
+    decimal = 1
+})
+table.insert(ROLE_CONVARS[ROLE_SOULBOUND], {
+    cvar = "ttt_soulbound_beebarrel_bees",
+    type = ROLE_CONVAR_TYPE_NUM,
+    decimal = 0
+})
+
+if SERVER then
+    local function BeeBarrelDamage(target, dmginfo)
+        if target:GetClass() ~= "prop_physics" then return end
+        if dmginfo:GetDamage() < 1 then return end
+
+        local model = target:GetModel()
+        if model ~= "models/bee_drum/beedrum002_explosive.mdl" then return end
+
+        local pos = target:GetPos()
+        timer.Create("TTTSoulboundBeeBarrelSpawn",0.1,beebarrel_bees:GetInt(),function()
+            local spos = pos + Vector(math.random(-50, 50), math.random(-50, 50), math.random(0, 100))
+            local headBee = ents.Create("npc_manhack")
+            headBee:SetPos(spos)
+            headBee:Spawn()
+            headBee:Activate()
+            headBee:SetNPCState(NPC_STATE_ALERT)
+            if scripted_ents.Get("ttt_beenade_proj") ~= nil then
+                local bee = ents.Create("prop_dynamic")
+                bee:SetModel("models/lucian/props/stupid_bee.mdl")
+                bee:SetPos(spos)
+                bee:SetParent(headBee)
+                headBee:SetNoDraw(true)
+            end
+            headBee:SetHealth(10)
+        end)
+    end
+
+    function ABILITY:Bought(soulbound)
+        soulbound:SetNWInt("TTTSoulboundBeeBarrelUses", beebarrel_uses:GetInt())
+        soulbound:SetNWFloat("TTTSoulboundBeeBarrelNextUse", CurTime())
+        hook.Add("EntityTakeDamage", "TTTSoulboundBeeBarrelDamage", BeeBarrelDamage)
+    end
+
+    function ABILITY:Condition(soulbound, target)
+        if not soulbound:IsInWorld() then return false end
+        if beebarrel_uses:GetInt() > 0 and soulbound:GetNWInt("TTTSoulboundBeeBarrelUses", 0) <= 0 then return false end
+        if CurTime() < soulbound:GetNWFloat("TTTSoulboundBeeBarrelNextUse") then return false end
+        return true
+    end
+
+    function ABILITY:Use(soulbound, target)
+        local plyPos = soulbound:GetPos()
+        local hitPos = soulbound:GetEyeTrace().HitPos
+        local vec = hitPos - plyPos
+        local fwd = Vector(0, 0, 0)
+        if target then
+            fwd = soulbound:GetForward() * 48
+            vec = Vector(0, 0, -1)
+        end
+        local spawnPos = hitPos - (vec:GetNormalized() * 15) + fwd
+
+        local ent = ents.Create("prop_physics")
+        ent:SetModel("models/bee_drum/beedrum002_explosive.mdl")
+        ent:SetPos(spawnPos)
+        ent:Spawn()
+
+        local uses = soulbound:GetNWInt("TTTSoulboundBeeBarrelUses", 0)
+        uses = math.max(uses - 1, 0)
+        soulbound:SetNWInt("TTTSoulboundBeeBarrelUses", uses)
+        soulbound:SetNWFloat("TTTSoulboundBeeBarrelNextUse", CurTime() + beebarrel_cooldown:GetFloat())
+    end
+
+    function ABILITY:Cleanup(soulbound)
+        soulbound:SetNWInt("TTTSoulboundBeeBarrelUses", 0)
+        soulbound:SetNWFloat("TTTSoulboundBeeBarrelNextUse", 0)
+    end
+end
```

Comment:
We need to check if the bee grenade exists and if it doesnt this ability needs to be disabled. See lines 50-56 in packapunch.lua for example

Reply:
done